relational/table.py

```diff
@@ -230,58 +230,72 @@ def __init__(self):
         ])
 
     def _gen_facts(self, row, proj):
-        pts_pred = [('permit_type', lambda x: x == '1' or x == '2')]
-        if proj.field('address', PPTS.NAME) != '':
-            row[self.index(self.ADDRESS)] = proj.field('address', PPTS.NAME)
+        used_mohcd = False
+        for mohcd in _MOHCD_TYPES.keys():
+            if used_mohcd or proj.field('project_id', mohcd) == '':
+                continue
+
+            used_mohcd = True
+
+            name = proj.field('project_name', mohcd)
+
+            num = proj.field('street_number', mohcd)
+            addr = proj.field('street_name', mohcd)
+            if num:
+                addr = ('%s %s' % (num, addr))
+
+            if name:
+                addr = ('%s, %s' % (name, addr))
+
+            row[self.index(self.ADDRESS)] = '%s %s, %s' % (
+                    addr,
+                    proj.field('street_type', mohcd),
+                    proj.field('zip_code', mohcd))
+            sponsor = proj.field('project_lead_sponsor', mohcd)
+            if not sponsor:
+                sponsor = proj.field('project_sponsor', mohcd)
+            row[self.index(self.APPLICANT)] = sponsor
+
+            row[self.index(self.SUPERVISOR_DISTRICT)] = \
+                proj.field('supervisor_district', mohcd)
+
+            row[self.index(self.PERMIT_AUTHORITY_ID)] = proj.fk(mohcd)
+            row[self.index(self.PERMIT_AUTHORITY)] = 'mohcd'  # TODO
+
+        if used_mohcd:
+            return
+
+        addr = proj.field('address', PPTS.NAME)
+        if not addr:
+            addr = proj.field('name', PPTS.NAME)
```

Comment:
What's the harm in having an empty address? It feels like an empty address might be better than a 'project description' type of text in the address?

Reply:
the harm is that at least right now there's no way to identify what the project is in a project listing in the dashboard. they are only differentiable by their unit counts essentially (well, and their id)

Reply:
But i thought that the project would have an entry in ProjectFacts as well as the other tables it's just it's entry in ProjectFacts would have a null address? Or am I misunderstanding this?

Reply:
the project would have an entry in project facts, but the address is the primary way to identify it. it may even have a sponsor and supervisor district, but it shows up as blank otherwise.

Reply:
Let's not block changes any further for now, I guess I'm just concerned that project names typically have a hodge-podge of things which may break assumptions we make on what addresses look like, but it's not a huge deal so lgtm